### PR TITLE
fully handle 8->88 decays

### PR DESCRIPTION
**Pull request by wkotlarski**
@Expander, it wasn't long before I hit a problem with my design. As a reminder, we are tagging colour through the enum
```cpp
enum class ParticleColorRep {
   singlet,
   triplet,  
   anti_triplet,
   sextet,
   octet
};
```
As I understand, because we want to specialize on enum value, this is conceptually the same as specializing on value (like 5) and not on type (like int). We then want to template 
```cpp
struct Decay_amplitude_SSS {    
   ...
   std::complex<double> form_factor {0.};    
      
   double square() const;    
   ...    
};
```
because `form_factor` and  `square()` member must know about the colour. So far so good. The problem is where do the `Decay_amplitude_SSS` come from. They are computed by functions like `calculate_diagram_SSS_t1g3n3_UUU`. Such function returns `Decay_amplitude_SSS`. But `calculate_diagram_SSS_t1g3n3_UUU` doesn't need to know about colour. The reason is that for now we assume that every vertex is of the form lorentz * colour. So every single diagram contributes only lorentz * colour. Only when we add diagrams together we need to keep track that we add them as vectors in colour space.

In short, I don't need templified `calculate_diagram_*`. One idea is make `calculate_diagram_*` always return
```cpp
Decay_amplitude_SSS<ParticleColorRep::singlet, ParticleColorRep::singlet, ParticleColorRep::singlet>
```
and change the plus operator for `Decay_amplitude_*` such that it's possible to add it to
```cpp
Decay_amplitude_SSS<ParticleColorRep colIn1, ParticleColorRep colOut1, ParticleColorRep colOu2>
```
with only a special treatment for adding to
```cpp
Decay_amplitude_SSS<ParticleColorRep::octet, ParticleColorRep::octet, ParticleColorRep::octet>
```

### CODE REVIEW

**Comment by Expander** [comment]
I think your proposal makes sens. However, I don't understand the following:

You're saying that the function `calculate_diagram_*` calculates only the Lorentz part of the amplitude (which can be interpreted as a color-singlet, as you said). The returned object is of type `Decay_amplitude_SSS` etc. This returned object does not include the color information then, of course.

Now, where and how does the color information come into play?

Very naively, I'd have thought that a natural structure could look like so:

* The function `calculate_diagram_*` returns an object of type `Decay_amplitude_SSS_lorentz_part` etc.
* From this object, a new object is created, which combines this Lorenz part with the color information:
~~~.cpp
struct Decay_amplitude_SSS {
   Decay_amplitude_SSS_lorentz_part lorentz_part;
   Decay_amplitude_SSS_color_part color_part; // this could be templated
};

// calculate Lorentz part
Decay_amplitude_SSS_lorentz_part lorentz_part = calculate_diagram_*(...);

// calculate color part
Decay_amplitude_SSS_color_part color_part = ??? // where does this come from?

// create the amplitude
Decay_amplitude_SSS amp{lorentz_part, color_part};
~~~
In this naive setup only the `color_part` would need to be templated.

However, I think this naive thought misses some point. So, how would your new approach compare to this naive approach? How does the color structure come into play? Sorry, if this is a dump question.
